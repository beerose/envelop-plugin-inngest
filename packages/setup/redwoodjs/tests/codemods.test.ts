const path = require('path');
const fs = require('fs');
const jscodeshift = require('jscodeshift');
const pluginTransform = require('../src/use-inngest-codemod');
const functionTransform = require('../src/add-function-to-inngest-handler-codemod');

describe('codemods', () => {
  const defaultJscodeshiftOpts = {
    verbose: 1,
    dry: false,
    print: true,
    babel: false,
    extensions: ['js', 'json'],
    ignorePattern: '**/node_modules/**',
    ignoreConfig: [],
    runInBand: false,
    silent: false,
    parser: 'ts',
    parserConfig: {},
    failOnError: true,
    stdin: false,
  };

  describe('plugin', () => {
    describe('When using use-inngest-codemod to transform, the graphQL handler', () => {
      describe('transforms code correctly', () => {
        it('when given a standard default GraphQL handler', () => {
          const source = fs.readFileSync(
            path.join(__dirname, '__testfixtures__', 'plugin', 'default.input.js'),
            'utf8'
          );
          const output = pluginTransform(
            { path: '../__testfixtures__/default.input.js', source },
            { jscodeshift },
            { ...defaultJscodeshiftOpts }
          );
          expect(output).toMatchSnapshot();
        });

        it('when given a standard default GraphQL handler with dbAuth setup', () => {
          const source = fs.readFileSync(
            path.join(__dirname, '__testfixtures__', 'plugin', 'default.input.js'),
            'utf8'
          );
          const output = pluginTransform(
            { path: '../__testfixtures__/dbauth.input.js', source },
            { jscodeshift },
            { ...defaultJscodeshiftOpts }
          );
          expect(output).toMatchSnapshot();
        });

        it('when GraphQL handler already configures some extraPlugins', () => {
          const source = fs.readFileSync(
            path.join(__dirname, '__testfixtures__', 'plugin', 'extra-plugins.input.js'),
            'utf8'
          );
          const output = pluginTransform(
            { path: '../__testfixtures__/extra-plugins.input.js', source },
            { jscodeshift },
            { ...defaultJscodeshiftOpts }
          );

          expect(output).toMatchSnapshot();
        });

        it('when GraphQL handler already has useInngest setup', () => {
          const source = fs.readFileSync(path.join(__dirname, '__testfixtures__', 'plugin', 'exists.input.js'), 'utf8');
          const output = pluginTransform(
            { path: '../__testfixtures__/exists.input.js', source },
            { jscodeshift },
            { ...defaultJscodeshiftOpts }
          );
          expect(output).toMatchSnapshot();
        });
      });
    });
  });
  describe('function', () => {
    describe('When using add-function-to-inngest-handler-codemod to transform, the Inngest handler', () => {
      describe('transforms code correctly', () => {
        describe('when given a standard default Inngest handler', () => {
          describe('it adds the new function to the existing array of inngest functions', () => {
            const source = fs.readFileSync(
              path.join(__dirname, '__testfixtures__', 'function', 'default.input.js'),
              'utf8'
            );
            const output = functionTransform(
              { path: '../__testfixtures__/default.input.js', source },
              { jscodeshift },
              { ...defaultJscodeshiftOpts, functionName: 'testFunction' }
            );
            expect(output).toMatchSnapshot();
          });
        });
      });
      describe('when given an Inngest handler with an empty functions array', () => {
        describe('it adds the new function to the empty array of inngest functions', () => {
          const source = fs.readFileSync(
            path.join(__dirname, '__testfixtures__', 'function', 'empty.input.js'),
            'utf8'
          );
          const output = functionTransform(
            { path: '../__testfixtures__/empty.input.js', source },
            { jscodeshift },
            { ...defaultJscodeshiftOpts, functionName: 'addedFunctionToEmpty' }
          );
          expect(output).toMatchSnapshot();
        });
      });
    });
  });
});
