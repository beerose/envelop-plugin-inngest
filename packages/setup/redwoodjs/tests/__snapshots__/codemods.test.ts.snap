// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"import { serve } from 'inngest/redwood';

import { INNGEST_APP_NAME } from 'src/inngest/client';
import testFunction from \\"src/inngest/testFunction\\";
import helloWorld from 'src/inngest/helloWorld';

const inngestFunctions = [helloWorld, testFunction];

export const handler = serve(INNGEST_APP_NAME, inngestFunctions, {
  servePath: '/inngest',
  // signingKey: 'Redwood',
});
"
`;

exports[` 2`] = `
"import { serve } from 'inngest/redwood';

import { INNGEST_APP_NAME } from 'src/inngest/client';

import addedFunctionToEmpty from \\"src/inngest/addedFunctionToEmpty\\";

const inngestFunctions = [addedFunctionToEmpty];

export const handler = serve(INNGEST_APP_NAME, inngestFunctions, {
  servePath: '/inngest',
  // signingKey: 'Redwood',
});
"
`;

exports[`codemods plugin When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when GraphQL handler already configures some extraPlugins 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server';

import directives from 'src/directives/**/*.{js,ts}';
import sdls from 'src/graphql/**/*.sdl.{js,ts}';
import services from 'src/services/**/*.{js,ts}';

import { getCurrentUser } from 'src/lib/auth';
import { db } from 'src/lib/db';
import { logger } from 'src/lib/logger';

import { foo } from 'src/lib/foo';

import { inngestPlugin } from \\"src/plugins/inngest\\";

export const handler = createGraphQLHandler({
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  extraPlugins: [foo, inngestPlugin],

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  }
});
"
`;

exports[`codemods plugin When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when GraphQL handler already has useInngest setup 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server';

import directives from 'src/directives/**/*.{js,ts}';
import sdls from 'src/graphql/**/*.sdl.{js,ts}';
import services from 'src/services/**/*.{js,ts}';

import { getCurrentUser } from 'src/lib/auth';
import { db } from 'src/lib/db';
import { logger } from 'src/lib/logger';

import { inngestPlugin } from 'src/inngest/plugin';

export const handler = createGraphQLHandler({
  getCurrentUser,
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,
  extraPlugins: [inngestPlugin],

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  }
});
"
`;

exports[`codemods plugin When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when given a standard default GraphQL handler 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server';

import directives from 'src/directives/**/*.{js,ts}';
import sdls from 'src/graphql/**/*.sdl.{js,ts}';
import services from 'src/services/**/*.{js,ts}';

import { db } from 'src/lib/db';
import { logger } from 'src/lib/logger';

import { inngestPlugin } from \\"src/plugins/inngest\\";

export const handler = createGraphQLHandler({
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  },

  extraPlugins: [inngestPlugin]
});
"
`;

exports[`codemods plugin When using use-inngest-codemod to transform, the graphQL handler transforms code correctly when given a standard default GraphQL handler with dbAuth setup 1`] = `
"import { createGraphQLHandler } from '@redwoodjs/graphql-server';

import directives from 'src/directives/**/*.{js,ts}';
import sdls from 'src/graphql/**/*.sdl.{js,ts}';
import services from 'src/services/**/*.{js,ts}';

import { db } from 'src/lib/db';
import { logger } from 'src/lib/logger';

import { inngestPlugin } from \\"src/plugins/inngest\\";

export const handler = createGraphQLHandler({
  loggerConfig: { logger, options: {} },
  directives,
  sdls,
  services,

  onException: () => {
    // Disconnect from your database with an unhandled exception.
    db.$disconnect();
  },

  extraPlugins: [inngestPlugin]
});
"
`;
